components/VM_Arm/src/main.c

```diff
@@ -621,13 +621,36 @@ static void irq_handler(void *data, ps_irq_acknowledge_fn_t acknowledge_fn, void
     }
 }
 
+/* Default RAM initialization. Modules can overwrite this weak function with a
+ * custom initialization, e.g. to use memory shared with other components or
+ * mapped on demand.
+ */
+WEAK int vm_init_ram(vm_t *vm, const vm_config_t *vm_config)
```

Comment:
I see you renamed `init_ram_module()` to `vm_init_ram()`. This means at least `vm_introspect` example in https://github.com/seL4/camkes-vm-examples gets broken (it is broken currently for other reasons, but that's another story). Perhaps some code we don't know of.

Personally I'm not against breaking the ABI but others might be.

Reply:
The different name is intentional, it's based on you last comment that changing the signature of an existing function is not a good idea. It this PR gets merged, I will create an issue for `vm_introspect` so this gets adapted.

Reply:
> [`vm_introspect`] is broken currently for other reasons, but that's another story). Perhaps some code we don't know of.

I think https://github.com/seL4/camkes-vm-examples/pull/39 will fix the brokenness you are referring to.

Reply:
Good spotting but not exactly: at some point CAmkES started using as large as possible frames for dataports. Hence if you had more than 2 MB dataport, it uses ARM large pages. On the other hand, currently VMM always uses 4 kB pages in `vm_ram_touch()`. To be able to use 2 MB pages from dataport as guest RAM, we use this kind of hack in tiiuae/sel4_projects_libs@fa99679290c03deba72b18ac39b491599eb63d31 .

Reply:
Forgot to add that you need to use large pages here as well: https://github.com/seL4/camkes-vm-examples/blob/master/apps/Arm/vm_introspect/src/init_dataport_ram.c

Reply:
@hlyytine: Could you create an issue in https://github.com/seL4/camkes-vm-examples to track this?